### PR TITLE
Subscriber guest is not removed from email_contact

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

**Comment by cdiacon** [comment]
@adeelq this ^^^ will create imports "Contact_Delete" but  import type "Contact_Delete" status is never marked as imported.

**Comment by adeelq** [comment]
@cdiacon https://github.com/dotmailer/dotmailer-magento2-extension/blob/master/Model/Sync/Contact/Delete.php#L54 will set to imported

**Comment by cdiacon** [comment]
@adeelq k found the issue - 
https://github.com/dotmailer/dotmailer-magento2-extension/blob/653c47e5017bdf8be3df057f9a62b87abdc7b77d/Model/Sync/Contact/Delete.php#L31-L31

delete method will have always "null" repsonsebody - you should check for $result->message instead so the single sync can be proccessed!

**Comment by cdiacon** [comment]
issue moved to https://github.com/dotmailer/dotmailer-magento2-extension/issues/246

File: Observer/Newsletter/RemoveContact.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        \Dotdigitalgroup\Email\Model\ContactFactory $contactFactory,
+        \Dotdigitalgroup\Email\Helper\Data $data,
+        \Magento\Store\Model\StoreManagerInterface $storeManagerInterface,
+        \Dotdigitalgroup\Email\Model\ImporterFactory $importerFactory
+    )
+    {
+        $this->contactFactory = $contactFactory;
+        $this->helper = $data;
+        $this->storeManager = $storeManagerInterface;
+        $this->importerFactory = $importerFactory->create();
```
Comment (cdiacon) [inline Observer/Newsletter/RemoveContact.php]: @adeelq importerFactory means that I can always create new instance in the code bellow, but doing this in constructor becomes a non factory model.

**Pull request merged**